File: ex/loop3-1/loop3-1.c

int x=0;
int y=50;
int main() {
while(x<100){
	if(x+1>50)
	{
		x=x+1;
		y=y+1;
	}
	else
	{
		x=x+1;
	}
}
assert(y==100);
}

File: ex/loop3-1/loop3-1.m
type number : 0..200;
var x : number;
	y : number;
	res_y : number;


startstate "Init"
begin
	x := 0;
	y := 50;
	res_y := 100;
endstartstate;

rule "loop"
	x < 100 ==> begin
	if(x + 1 > 50)then
		x := x + 1;
		y := y + 1;
	else
		x := x + 1;
	end;
	endrule;

rule "assert"
	x >= 100 ==> begin
	res_y := y;
	endrule;
	
	
	
invariant "loop3"
  res_y = 100;